src/http/http2_parser.h

```diff
@@ -61,8 +68,7 @@ namespace http2
 
       // Submit initial settings
       std::vector<nghttp2_settings_entry> settings;
-      settings.push_back({NGHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS, 1});
```

Comment:
Removed the limitation on the maximum number of concurrent streams, as already done in #4715